src/extension/ddappsec.c

```diff
@@ -328,7 +357,10 @@ int dd_appsec_rshutdown()
     }
 
     dd_helper_rshutdown();
-    dd_tags_rshutdown();
+
+    if (DDAPPSEC_G(enabled) == ENABLED) {
+        dd_tags_rshutdown();
```

Comment:
If `request_init` was sent, followed by a `config_features` disabling appsec, you will be leaking some memory by not calling `dd_tags_rshutdown`.

Reply:
If you were meaning `_appsec_json_frags`, I have refactored it now so it gets cleaned